verification/spec/relayer/Environment.tla

```diff
@@ -0,0 +1,334 @@
+      
+    - channelID -- a channel identifier
+      Stores the channel identifier of this channel end.  
+    
+    - counterpartyChannelID -- a channel identifier
+      Stores the channel identifier of the counterparty channel end.
+      
+      (** ignoring channel ordering for now **) 
+ ***************************************************************************)   
+ChannelEnds ==
+    [
+        state : ChannelStates, 
+        channelID : ChannelIDs \union {nullChannelID},
+        counterpartyChannelID : ChannelIDs \union {nullChannelID}
+    ] 
+
+(***************************** ConnectionEnds *****************************
+    A set of connection end records. 
+    A connection end record contains the following fields:
+    
+    - state -- a string 
+      Stores the current state of this connection end. It has one of the 
+      following values: "UNINIT", "INIT", "TRYOPEN", "OPEN".
+    
+    - connectionID -- a connection identifier
+      Stores the connection identifier of this connection end.
+    
+    - counterpartyConnectionID -- a connection identifier
+      Stores the connection identifier of the counterparty connection end.
+    
+    - clientID -- a client identifier
+      Stores the client identifier associated with this connection end. 
+      
+    - counterpartyClientID -- a client identifier
+      Stores the counterparty client identifier associated with this connection end.
+ ***************************************************************************)
+ConnectionEnds ==
+    [
+        state : ConnectionStates,
+        connectionID : ConnectionIDs \union {nullConnectionID},
+        clientID : ClientIDs \union {nullClientID},
+        counterpartyConnectionID : ConnectionIDs \union {nullConnectionID},
+        counterpartyClientID : ClientIDs \union {nullClientID}, 
+        channelEnd : ChannelEnds
+    ]      
+
+(********************************** Chains *********************************
+    A set of chain records. 
+    A chain record contains the following fields:
+    
+    - height : an integer between nullHeight and MaxHeight. 
+      Stores the current height of the chain.
+    
+    - counterpartyClientHeight : an integer between nullHeight and MaxHeight
+      Stores the height of the client for the counterparty chain.
+
+    - connectionEnd : a connection end record 
+      Stores data about the connection with the counterparty chain
+ ***************************************************************************)
+Chains ==    
+    [
+        height : Heights \union {nullHeight},
+        counterpartyClientHeight : Heights \union {nullHeight},
+        connectionEnd : ConnectionEnds
+    ]
+    
+(******************************** Datagrams ********************************
+ A set of datagrams.
+ ***************************************************************************)
+Datagrams ==
+    [type : {"CreateClient"}, clientID : ClientIDs, height : Heights]
```

Comment:
One question here:
Why is the `type` always a set? I understand everything is a set in TLA+. But could we also write `type : "CreateClient"` here and get the same behavior?